### PR TITLE
Add HttpSenderProvider SPI

### PR DESCRIPTION
**Pull request by jack-berg**
Followup to #5505. Related to #5351.

This adds a new HttpSenderProvider SPI interface for configuring different HttpSender implementations.

It also splits out the OkHttpSender into a new artifact `opentelemetry-exporter-http-sender-okhttp`.

Another followup PR will define a JDK based HttpSender in a new artifact `opentelemetry-exporter-http-sender-jdk`.

### CODE REVIEW

File: exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporterBuilder.java [hunk 1]
```diff
@@ -125,17 +129,27 @@ public HttpExporter<T> build() {
...
-            compressionEnabled,
-            exportAsJson ? "application/json" : "application/x-protobuf",
-            timeoutNanos,
-            headerSupplier,
-            authenticator,
-            retryPolicy,
-            tlsConfigHelper.getSslContext(),
-            tlsConfigHelper.getTrustManager());
+    HttpSender httpSender = null;
+    for (HttpSenderProvider httpSenderProvider :
```
Comment (brunobat) [inline exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporterBuilder.java:133]: Is there a case for more than one sender provider?
Should we force to have just one? I think in most cases, having 2 of them is more of a bug due to conflicting dependencies.

Comment (mateuszrzeszutek) [reply exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporterBuilder.java:133]: Could we perhaps print a warning when >1 sender implementations are found on classpath?

File: exporters/http-sender/okhttp/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  id("otel.java-conventions")
+  id("otel.publish-conventions")
+
+  id("otel.animalsniffer-conventions")
+}
+
+description = "OpenTelemetry OkHttp HttpSender"
+otelJava.moduleName.set("io.opentelemetry.exporter.http.okhttp.internal")
+
+val versions: Map<String, String> by project
```
Comment (mateuszrzeszutek) [inline exporters/http-sender/okhttp/build.gradle.kts:11]: `versions` does not seem to be used anywhere

**Review by mateuszrzeszutek — APPROVED** [review]

File: exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporterBuilder.java (hunk 1 above)
Comment (brunobat) [reply exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporterBuilder.java:133]: That's fine, but the question remains. 
> Is there a case for more than one sender provider?

Comment (brunobat) [reply exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporterBuilder.java:133]: This is reasonable to me.

**Review by brunobat — APPROVED** [review]
Thanks @jack-berg !

File: exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporterBuilder.java (hunk 1 above)
Comment (jack-berg) [reply exporters/common/src/main/java/io/opentelemetry/exporter/internal/http/HttpExporterBuilder.java:134]: Added a TODO for now which I'll resolve in the next PR.

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by jkwatson — APPROVED** [review]

**Pull request merged**